DonoGear/app/src/main/java/com/example/donogear/actionpages/ProductDetails.java

```diff
@@ -0,0 +1,314 @@
+
+public class ProductDetails extends AppCompatActivity {
+
+    private String itemId, itemName, itemDescription, itemHighestBidder;
+    private int itemBidAmount;
+    private Date itemTime;
+    private List<File> itemImages;
+    private List<String> itemVideosUrl;
+    private ItemProceedsDetails proceedsDetails;
+    private ParseQuery<ParseObject> itemVideosQuery, itemProceedsQuery;
+    private Handler handler;
+    private Context context;
+
+    private boolean hasVideos, hasProceeds;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_product_details);
+
+        initData();
+        itemVideosUrl = getItemVideos();
+        getItemProceedsDetails();
+        LinearLayout horizontalScrollViewContainer = findViewById(R.id.inner_layout);
+        LinearLayout itemVideosLayout = findViewById(R.id.videoButtons);
+        displayImages(itemImages, horizontalScrollViewContainer);
+        displayRemainingTime();
+        displayItemDetails();
+        Runnable videoRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if(hasVideos) {
+                    displayVideo(itemVideosUrl, itemVideosLayout);
+                }
+                else {
+                    System.out.println("Video? : " + hasVideos);
+                    handler.postDelayed(this, 1000);
+                }
+            }
+        };
+        handler.post(videoRunnable);
+
+        Runnable proceedsRunnable = new Runnable() {
+            @Override
+            public void run() {
+                if(hasProceeds) {
+                    displayProceedsDetails();
+                }
+                else {
+                    handler.postDelayed(this, 1000);
+                }
+            }
+        };
+        handler.post(proceedsRunnable);
+    }
+
+    private void initData() {
+        handler = new Handler();
+        context = getBaseContext();
+
+        Intent intent = getIntent();
+        ItemDetails itemDetails = (ItemDetails) intent.getSerializableExtra("item_details");
+        itemId = itemDetails.id;
+        itemName = itemDetails.itemName;
+        itemDescription = itemDetails.itemDescription;
+        itemImages = itemDetails.listOfImages;
+        itemTime = itemDetails.endDate;
+        itemHighestBidder = itemDetails.highestBidder;
+        itemBidAmount = itemDetails.currentPrice;
+
+        ImageButton back =  findViewById(R.id.backbtn);
+        back.setOnClickListener(view -> finish());
+        hasProceeds = false;
+        hasVideos = false;
+    }
+
+    /**
+     * Displays the video snipper
+     * TODO - Working code for snippet / better UI to show video display button
+     */
+    private void displayVideo(List<String> videoList, LinearLayout videoContainer) {
+//        final VideoView videoView = findViewById(R.id.videoView);
```

Comment:
Should remove the commented out code here and the lines below if they are not needed